sdk/go/internal/engineconn/session.go

```diff
@@ -18,15 +19,19 @@ import (
 type cliSessionConn struct {
 	*http.Client
 	childStdin io.Closer
+	childProc  *exec.Cmd
 }
 
 func (c *cliSessionConn) Host() string {
 	return "dagger"
 }
 
 func (c *cliSessionConn) Close() error {
-	if c.childStdin != nil {
-		return c.childStdin.Close()
+	if c.childStdin != nil && c.childProc != nil {
+		return errors.Join(
+			c.childStdin.Close(),
+			c.childProc.Wait(),
```

Comment:
If we leave this in we should add a timeout to the wait (there's a new(?) `.WaitDelay` field in `exec.Command` that seems to make this easy), just to protect against any bugs or other weirdness from being able to block Close indefinitely.